hoot-josm/src/main/cpp/hoot/core/josm/ops/JosmMapValidatorAbstract.cpp

```diff
@@ -0,0 +1,293 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2019, 2020 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+#include "JosmMapValidatorAbstract.h"
+
+// hoot
+#include <hoot/core/util/Log.h>
+#include <hoot/core/jni/JavaEnvironment.h>
+#include <hoot/core/util/StringUtils.h>
+#include <hoot/core/jni/JniConversion.h>
+#include <hoot/core/util/MapProjector.h>
+#include <hoot/core/jni/JniUtils.h>
+
+namespace hoot
+{
+
+JosmMapValidatorAbstract::JosmMapValidatorAbstract() :
+_maxElementsForMapString(2000000),
+_javaEnv(JavaEnvironment::getEnvironment()),
+_josmInterfaceInitialized(false),
+_numValidationErrors(0),
+_numFailingValidators(0)
+{
+  _josmInterfaceName = ConfigOptions().getJosmMapValidatorJavaImplementation();
+}
+
+JosmMapValidatorAbstract::~JosmMapValidatorAbstract()
+{
+  // See related note in _initJosmImplementation.
+  //if (_javaEnv != 0 && _josmInterface != 0)
+  //{
+    // We have a memory leak if this doesn't happen.
```

Comment:
Is this taken care of elsewhere?

Reply:
I'm going to go ahead and take out that commented out block. To avoid confusion. There's a comment ~line 85 explaining things. The JNI docs led me to believe I needed to do that cleanup there in the destructor but when I do it crashes every time. I did test without that cleanup against some larger input files and didn't notice any memory leaks when the jobs ended...but of course will need to eventually profile everything to be sure.